hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/prehandle/PreHandleContextImpl.java

```diff
@@ -290,17 +290,21 @@ public PreHandleContext requireKeyOrThrow(
         if (account == null) {
             throw new PreCheckException(responseCode);
         }
-
+        // If it is hollow account, and we require this to sign, we need to finalize the account
+        // with the corresponding ECDSA key in handle
+        if (isHollow(account)) {
+            // Verify this key isn't for an immutable account
+            verifyNotStakingAccounts(account.accountIdOrThrow(), responseCode);
```

Comment:
⚠️ [`isHollow()`](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/HapiUtils.java#L94) has been changed to require account num > 1000,
```
        return (account.accountIdOrThrow().accountNum() > 1000
                && account.keyOrElse(EMPTY_KEY_LIST).equals(EMPTY_KEY_LIST)
                ...
```
which means now `isHollow(0.0.800) == false` and this `verifyNotStakingAccounts()` is a no-op.

So it seems we should move `verifyNotStakingAccounts()` below this `if` block.
```
if (isHollow(account)) {
    requiredHollowAccounts.add(account);
    return this;
}
verifyNotStakingAccounts(account.accountIdOrThrow(), responseCode);

```

Reply:
Yes, makes sense. Thanks @tinker-michaelj . Changed.